### PR TITLE
[Shop][Product] Display product original price on product page

### PR DESCRIPTION
**Pull request by JulienLoison**
- [ ] Document the BC breaks

| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | yes
| Deprecations?   | no
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/8258
| License         | MIT

When original price is greater than the actual price of the product, it is displayed on the product page. 

image

### CODE REVIEW

File: src/Sylius/Behat/Page/Shop/Product/ShowPageInterface.php [hunk 1]
```diff
@@ -54,6 +54,8 @@ public function getName(): string;
 
     public function getPrice(): string;
 
+    public function getOriginalPrice(): string;
+
```
Comment (gabiudrescu) [inline src/Sylius/Behat/Page/Shop/Product/ShowPageInterface.php]: this is a BC Break.

File: src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculatorInterface.php [hunk 2]
```diff
@@ -22,4 +22,9 @@ interface ProductVariantPriceCalculatorInterface
      * @throws MissingChannelConfigurationException when price for given channel does not exist
      */
     public function calculate(ProductVariantInterface $productVariant, array $context): int;
+
```
Comment (gabiudrescu) [inline src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculatorInterface.php]: this is a BC Break.

**Comment by JulienLoison** [comment]
@gabiudrescu I see what you expect on listings and widgets. This wouldn't be hard to implement by reusing this snippet 

`{% set hasDiscount = variant|sylius_has_discount({'channel': sylius.channel}) %}

{% if hasDiscount %}
    <span class="ui small header" id="product-original-price" {{ sylius_test_html_attribute('product-original-price', money.calculateOriginalPrice(variant)) }}>
        <del>{{ money.calculateOriginalPrice(variant) }}</del>
    </span>
{% endif %}`

This could be a first solution util a more elegant way is found ?

**Comment by gabiudrescu** [comment]
@JulienLoison I have nothing against this

File: src/Sylius/Behat/Page/Shop/Product/ShowPageInterface.php (hunk 1 above)
Comment (lchrusciel) [reply src/Sylius/Behat/Page/Shop/Product/ShowPageInterface.php]: Behats are not covered with our BC promise ;)

File: src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculatorInterface.php (hunk 2 above)
Comment (lchrusciel) [reply src/Sylius/Component/Core/Calculator/ProductVariantPriceCalculatorInterface.php]: That's true, we cannot merge this PR due to this change. Maybe just a separate service could do the job? Or a new one `ProductVariantPricesCalculatorInterface` (with s) and this one should be deprecated but used internally to ensure proper calculation logic

**Comment by lchrusciel** [comment]
Closing in favour of #11423

**Pull request closed without merging**